From 914d7d80f79c84390108647cc482706f30e2cf04 Mon Sep 17 00:00:00 2001
From: Nikesh Oswal <noswal@nvidia.com>
Date: Mon, 5 Nov 2012 17:53:54 +0530
Subject: [PATCH] audio: add ulp audio format query

Bug 998701

Change-Id: Ic3bc2a10541859f5bdc7d3824a5e9b237a725838

init: grant permission to AID_MEDIA for setting media properties

grant permission to AID_MEDIA for setting properties with .media
prefix

Bug: 998697
Change-Id: I1c2f3697ac1d593ff94346eaa3f4a486617541eb

audio: Add WFD device and WFD Enable Macro

Add WFD device and WFD Enable Macro to audio.h,
this defination will be used by nvcap driver,
libaudio and AudioPolicyManager

Bug: 987484
Change-Id: I6094aa6e39f9f4d81ff2d9e2f3b0a6200f914b54
Reviewed-on: http://git-psac/r/265
Tested-by: Nikesh Oswal <noswal@nvidia.com>
Reviewed-by: Scott Peterson <speterson@nvidia.com>
Reviewed-by: Isaac Richards <irichards@nvidia.com>

init: change UID of MediaServer to system

Temporarily change UID of MediaServer to system, so that
services can be launched from this process for WFD usecase

Change-Id: I3805a742c6e8d9c6d846e2504f7d59d5ae0fc079

audio: set sample size to 2 for IEC61937 format

Returning framesize of 1 for compressed stream affects timestamp
calculation in AudioPlayer.

Set sample size to 2 for compressed streams to fix timestamps.
Change AUDIO_FORMAT_AC3 to AUDIO_FORMAT_IEC61937 for generalizing.

Bug 998699

Reviewed-on: http://git-master/r/82665
(cherry picked from commit 35494f6b1891c90b7a79dbf08a8666d015a941ee)

Change-Id: I2b21b6825fc0d97521f6b1e58fe2e545296a0066

Revert : change UID of MediaServer to system

Merge the below commit after public release of MR1
service manager: make an entry for wfd audio service
http://git-master/r/#change,114458

Change-Id: I93f715e4ff389962acacf767e60673f5f948f98b

audio: introduction of ac3/dts capability flag

the ac3/dts capability flag is introduced to activate/deactive the
ac3/dts relevant code as per requirement

Bug 998699

Change-Id: Ie8ea7cb0dcb0e37cd70e05c6bee66db3963d310e

audio: add dual audio stream type

Bug 1054990

Change-Id: Iff8e7f34f2ba6d1419ede139ab6ae3c6ea4b4b0b
---
 include/system/audio.h  |   28 +++++++++++++++++++++++++++-
 init/property_service.c |    1 +
 2 files changed, 28 insertions(+), 1 deletions(-)

diff --git a/include/system/audio.h b/include/system/audio.h
index d246070..ddcab25 100644
--- a/include/system/audio.h
+++ b/include/system/audio.h
@@ -27,6 +27,10 @@
 
 __BEGIN_DECLS
 
+#define FRAMEWORK_HAS_WIFI_DISPLAY_SUPPORT 1
+#define FRAMEWORK_HAS_AC3_DTS_PASSTHROUGH_SUPPORT 1
+#define FRAMEWORK_HAS_DUAL_AUDIO_SUPPORT 1
+
 /* The enums were moved here mostly from
  * frameworks/base/include/media/AudioSystem.h
  */
@@ -46,7 +50,9 @@ typedef enum {
     AUDIO_STREAM_ENFORCED_AUDIBLE = 7, /* Sounds that cannot be muted by user and must be routed to speaker */
     AUDIO_STREAM_DTMF             = 8,
     AUDIO_STREAM_TTS              = 9,
-
+#ifdef FRAMEWORK_HAS_DUAL_AUDIO_SUPPORT
+    AUDIO_DUAL_AUDIO = 10,
+#endif
     AUDIO_STREAM_CNT,
     AUDIO_STREAM_MAX              = AUDIO_STREAM_CNT - 1,
 } audio_stream_type_t;
@@ -141,6 +147,7 @@ typedef enum {
     AUDIO_FORMAT_HE_AAC_V1           = 0x05000000UL,
     AUDIO_FORMAT_HE_AAC_V2           = 0x06000000UL,
     AUDIO_FORMAT_VORBIS              = 0x07000000UL,
+    AUDIO_FORMAT_IEC61937            = 0x08000000UL,
     AUDIO_FORMAT_MAIN_MASK           = 0xFF000000UL,
     AUDIO_FORMAT_SUB_MASK            = 0x00FFFFFFUL,
 
@@ -300,6 +307,9 @@ enum {
     AUDIO_DEVICE_OUT_USB_ACCESSORY             = 0x2000,
     AUDIO_DEVICE_OUT_USB_DEVICE                = 0x4000,
     AUDIO_DEVICE_OUT_REMOTE_SUBMIX             = 0x8000,
+#ifdef FRAMEWORK_HAS_WIFI_DISPLAY_SUPPORT
+    AUDIO_DEVICE_OUT_WIFI_DISPLAY              = 0x1000000,
+#endif
     AUDIO_DEVICE_OUT_DEFAULT                   = AUDIO_DEVICE_BIT_DEFAULT,
     AUDIO_DEVICE_OUT_ALL      = (AUDIO_DEVICE_OUT_EARPIECE |
                                  AUDIO_DEVICE_OUT_SPEAKER |
@@ -317,6 +327,9 @@ enum {
                                  AUDIO_DEVICE_OUT_USB_ACCESSORY |
                                  AUDIO_DEVICE_OUT_USB_DEVICE |
                                  AUDIO_DEVICE_OUT_REMOTE_SUBMIX |
+#ifdef FRAMEWORK_HAS_WIFI_DISPLAY_SUPPORT
+                                 AUDIO_DEVICE_OUT_WIFI_DISPLAY |
+#endif
                                  AUDIO_DEVICE_OUT_DEFAULT),
     AUDIO_DEVICE_OUT_ALL_A2DP = (AUDIO_DEVICE_OUT_BLUETOOTH_A2DP |
                                  AUDIO_DEVICE_OUT_BLUETOOTH_A2DP_HEADPHONES |
@@ -505,6 +518,17 @@ static inline audio_channel_mask_t audio_channel_in_mask_from_count(uint32_t cha
     }
 }
 
+static inline bool audio_is_ulp_support(audio_format_t format)
+{
+    switch (format & AUDIO_FORMAT_MAIN_MASK) {
+    case AUDIO_FORMAT_MP3:
+    case AUDIO_FORMAT_AAC:
+        return true;
+    default:
+        return false;
+    }
+}
+
 static inline bool audio_is_valid_format(audio_format_t format)
 {
     switch (format & AUDIO_FORMAT_MAIN_MASK) {
@@ -520,6 +544,7 @@ static inline bool audio_is_valid_format(audio_format_t format)
     case AUDIO_FORMAT_HE_AAC_V1:
     case AUDIO_FORMAT_HE_AAC_V2:
     case AUDIO_FORMAT_VORBIS:
+    case AUDIO_FORMAT_IEC61937:
         return true;
     default:
         return false;
@@ -541,6 +566,7 @@ static inline size_t audio_bytes_per_sample(audio_format_t format)
         size = sizeof(int32_t);
         break;
     case AUDIO_FORMAT_PCM_16_BIT:
+    case AUDIO_FORMAT_IEC61937:
         size = sizeof(int16_t);
         break;
     case AUDIO_FORMAT_PCM_8_BIT:
diff --git a/init/property_service.c b/init/property_service.c
index f46accd..df3be0d 100755
--- a/init/property_service.c
+++ b/init/property_service.c
@@ -96,6 +96,7 @@ struct {
     { "selinux."         , AID_SYSTEM,   0 },
     { "modem.",            AID_RADIO,   0},
     { "sf.lcd_density",   AID_SYSTEM,   0 },
+    { "media.",            AID_MEDIA,   0},
     { NULL, 0, 0 }
 };
 
-- 
1.7.1

